library/HTMLPurifier/URIFilter/SafeIframe.php

```diff
@@ -56,12 +56,16 @@ public function filter(&$uri, $config, $context)
         if (!($token && $token->name == 'iframe')) {
             return true;
         }
+
         // check if we actually have some whitelists enabled
-        if ($this->regexp === null) {
-            return false;
+        if ($this->regexp !== null) {
+            // actually check the whitelists
+            return preg_match($this->regexp, $uri->toString());
         }
-        // actually check the whitelists
-        return preg_match($this->regexp, $uri->toString());
+
+        // check if the host is in a whitelist for safe iframe hosts
+        $safeHosts = $config->get('URI.SafeIframeHosts');
+        return is_array($safeHosts) && isset($safeHosts[$uri->host]);
```

Comment:
It feels like this should be a conjunction; if both configs are set, they both should be applied

Reply:
@ezyang I see your point. I'm making the changes.

Reply:
Hi @ezyang, I have changed the code in according with your suggestion. Could you please check again?